File: files/2025-05-11/CSEC 11052025.R
library(tidyverse)
library(readxl)

path = "files/2025-05-11/Challenge 23.xlsx"
input = read_excel(path, range = "B2:D28")
test = read_excel(path, range = "F2:G6") %>%
  rename("Quarter" = 1)

result = input %>%
  mutate(Quarter = paste0("Q", quarter(Date))) %>%
  summarise(
    Production = sum(
      `Production (L)`[`Temp (°C)` >= -10 & `Temp (°C)` <= -5],
      na.rm = TRUE
    ),
    .by = Quarter
  )

all.equal(result, test)
#> [1] TRUE

File: files/2025-05-11/CSEC 11052025.py
import pandas as pd

path = "Challenge 23.xlsx"
input_data = pd.read_excel(path, usecols="B:D", skiprows=1, nrows=27)
test = pd.read_excel(path, usecols="F:G", skiprows=1, nrows=5)
test.columns = ["Quarter", "Production"]

input_data["Date"] = pd.to_datetime(input_data["Date"])
input_data["Quarter"] = "Q" + input_data["Date"].dt.quarter.astype(str)
result = (
    input_data.loc[input_data["Temp (°C)"].between(-10, -5)]
    .groupby("Quarter", as_index=False)["Production (L)"]
    .sum()
    .rename(columns={"Production (L)": "Production"})
)

print(result.equals(test))
